File: src/actions/createCovercryptKeyPair.ts
import { KmsClient, Policy } from "cloudproof_js";
import { BACKEND_URL } from "./backendConfig";
import { KeyPair } from "./types";

//
// Create Covercrypt Key Pair
//
export const createCovercryptKeyPair = async (
  kmsToken: string,
  policy: Policy,
  tags: string[] | undefined = undefined
): Promise<KeyPair> => {
  // KMS Client
  const client = new KmsClient(BACKEND_URL, kmsToken);

  const masterKeys = await client.createCoverCryptMasterKeyPair(policy, tags);
  const masterSecretKeyUID = masterKeys[0];
  const masterPublicKeyUID = masterKeys[1];

  return {
    masterSecretKeyUID,
    masterPublicKeyUID,
  };
};

File: src/actions/createCovercryptKeyPair.java
//
// Create Covercrypt Key Pair
//
public static void createCoverCryptMasterKeyPair(String kmsServerUrl, Optional<String> apiKey) {
  Policy policy = policy();
  KmsClient kmsClient = new KmsClient(kmsServerUrl, apiKey);
  String[] ids = kmsClient.createCoverCryptMasterKeyPair(policy);
}
